src/praisonai/praisonai/integrations/base.py

```
@@ -183,20 +183,27 @@ async def execute_async(self, cmd: List[str], timeout: Optional[int] = None) ->
         try:
-            stdout, stderr = await asyncio.wait_for(
-                proc.communicate(),
-                timeout=timeout
-            )
+            try:
+                stdout, stderr = await asyncio.wait_for(
+                    proc.communicate(),
+                    timeout=timeout
+                )
+            except asyncio.TimeoutError:
+                raise TimeoutError(f"Command timed out after {timeout}s: {' '.join(cmd)}")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)